apps/vscode/command_client/command_client.py

```diff
@@ -197,20 +197,33 @@ def run_command(
     return decoded_contents["returnValue"]
 
 
-def get_communication_dir_path():
-    """Returns directory that is used by command-server for communication
+command_server_directory = None
 
-    Returns:
-        Path: The path to the communication dir
-    """
-    suffix = ""
 
+# NB: See https://github.com/talonhub/community/issues/966 for why we do OS-specific temp dirs
+def get_platform_specific_communication_dir_path():
     # NB: We don't suffix on Windows, because the temp dir is user-specific
     # anyways
+    suffix = ""
     if hasattr(os, "getuid"):
         suffix = f"-{os.getuid()}"
+    if app.platform == "linux" or app.platform == "mac":
+        return Path("/tmp") / f"{actions.user.command_server_directory()}{suffix}"
```

Comment:
is this going to present a problem for snaps / linux flatpak? cc/ @auscompgeek

Reply:
It seems like it will probably be a problem: https://forum.snapcraft.io/t/accessing-tmp-from-snaps/22384/17 . It seems like the suggestion there and elsewhere if you want to have a shared folder is to use bind mounts into the home directory. If that's the case I'm not so sure there's going to be any folder that's reliable to use with snaps, without using some sort of trickery like the bind mount, which will have to be up to the user. That said I'm not super familiar with snaps or flatpaks, so may be missing something.

I would guess that right now vscode as a snap wouldn't actually work with command-server anyway? Since snaps get their own temporary directories, which will differ from what talon gets. So at the very least we're not breaking something that already works. But of course it would be nice to have it also work for snaps if possible.

Reply:
wouldn't that argue for using home directory or talon directory in that case?

Reply:
Ya you are right. If someone wants to use it in a ram disk an know they're not going to be using a snap, they can create a symbolic link to some other location.

Updated it to use "~/.talon/.comms" for now and dropped the suffix as everything will be per-user now anwyay. I think it makes sense to just put it in talon folder if we're going to go this route. Happy to change the folder name to whatever though.